### PR TITLE
add support for zstandard compression

### PR DESCRIPTION
**Pull request by benthor**
This minimal PR adds support for the `zstandard` compression algorithm that the wikipedia `zim` files appear to have switched to.

### CODE REVIEW

File: article.go [hunk 1]
```diff
@@ -178,13 +178,18 @@ func (a *Article) Data() ([]byte, error) {
...
 		var ok bool
+		var dec io.ReadCloser
 		if blob, ok = blobLookup(); !ok {
 			b, err := a.z.bytesRangeAt(start+1, end+1)
 			if err != nil {
 				return nil, err
 			}
 			bbuf := bytes.NewBuffer(b)
-			dec, err := NewXZReader(bbuf)
+			if compression == 5 {
```
Comment (akhenakh) [inline article.go:188]: could you make a switch here so it will help for readability and future compression?

File: zstd_reader.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	dec, err := zstd.NewReader(r)
+	if err != nil {
+		return nil, err
+	}
+	return &ZstdReader{dec}, nil
+}
+
+func (zr *ZstdReader) Close() error {
+	zr.Close()
+	return nil
```
Comment (akhenakh) [inline zstd_reader.go:23]: wy not `return  zr.Close()` ?

**Comment by akhenakh** [comment]
I'll merge as is then fix, thank you again

File: zstd_reader.go (hunk 2 above)
Comment (benthor) [reply zstd_reader.go:23]: well, `zr.Close()` doesn't return `error` itself. I just wanted to make it explicitly visible in the code that there won't ever be an error returned.

Comment (akhenakh) [reply zstd_reader.go:23]: Yep saw it while refactoring, thank you

**Pull request merged**